src/rpc/blockchain.cpp

```diff
@@ -670,6 +670,32 @@ static bool GetUTXOStats(CCoinsView *view, CCoinsStats &stats)
     return true;
 }
 
+UniValue pruneblockchain(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "pruneblockchain\n"
+            "\nArguments:\n"
+            "1. \"height\"       (int, required) The block height to prune up to.\n");
+
+    if (pruneMode != PRUNE_MANUAL) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Cannot prune via RPC unless in manual prune mode.");
```

Comment:
`RPC_METHOD_NOT_FOUND` seems better for this (already used for wallet RPCs when the wallet is not enabled)